### DISCUSSION TITLE
How to encode aggregated inputs?

### DISCUSSION THREAD

**Discussion by augustoteixeira** [cartesi/cips#17] [General]
We need an initial proposal for encoding aggregated inputs.

There is an initial draft already here:

https://github.com/cartesi/cips/blob/aggregation/cips/cip-5.md

And the main advantages in this proposal are:
- it allows for different encodings of transactions (some may have better compression)
- it allows for different encodings and methods for signature verification
- these can be changed independently, due to a rigid constraint between how they interface.

I would be happy to discuss any questions or suggestions on the proposal!

**Comment by guidanoli** [comment]
Why should Transactions Protocols' and Signature Protocols' names have the header padded with zeros?
In other words, why not `TP-ba23` instead of `TP-0000ba23` and `SP-bbf2` instead of `SP-0000bbf2`?

**Comment by augustoteixeira** [comment]
This is a good point. I will edit it, thank you!
